### PR TITLE
PluginManager: Fix patch level check.

### LINKED ISSUE #2335: PluginManager::isCompatibleAPI() Does not properly evaluate plugin declared API version number.
### Issue description

When attempting to load a plugin that declares a patch level specific number, PocketMine-MP will fail to properly evaluate that number if the plugin declares a lower minor level version than the current server version.  

Example: Attempting to load a plugin which declares api: 3.0.7 on PocketMine-MP 3.1.1 will result with the plugin not being loaded and an error message given.
`Could not load plugin <plugin_name>: Incompatible API version (plugin requires one of: 3.0.7)`

- Expected result: Any plugin declaring a patch level requirement, should load if the minor level requirement is exceeded, regardless of the patch level requirement.
Examples:
   - 3.0.0 plugin loads on Server 3.1.1
   - 3.0.1 plugin loads on Server 3.1.1
   - 3.0.2 plugin loads on Server 3.1.1
   - 3.0.3 plugin loads on Server 3.1.1

- Actual result: What actually happened? Plugins declaring a patch level requirement will not load if the declared patch level is higher than the server patch level, and the minor level of the server is higher than the plugin declared minor level. 
Examples:
   - 3.0.0 plugin loads on Server 3.1.1
   - 3.0.1 plugin loads on Server 3.1.1
   - 3.0.2 plugin does not load on Server 3.1.1
   - 3.0.3 plugin does not load on Server 3.1.1

### Steps to reproduce the issue

1. Attempt to start a PocketMine-MP server with a plugin that declares a specific patch level.
2. Make sure the server version has a minor level, higher than the minor level declared by the plugin.
3. Make sure the  server version has a patch level lower than the patch level declared by the plugin.
4 Plugin will fail to load.

### OS and versions

* PocketMine-MP: 3.0.0 - 3.1.1
* PHP: 7.2.5
* Server OS: Windows 10, Linux, Mac
* Game version: NA

### Plugins
- Test on a clean server without plugins: is the issue reproducible without any plugins loaded?

If the issue is **not** reproducible without plugins:
- Have you asked for help on our forums before creating an issue?
- Can you provide sample, *minimal* reproducing code for the issue? If so, paste it in the bottom section
- Paste your list of plugins here (use the 'plugins' command in PocketMine-MP)

### Crashdump, backtrace or other files
- Do not paste crashdumps into an issue - please use our Crash Archive at https://crash.pmmp.io for submitting crash reports to not spam the issue tracker. Add links to your reports in the Crash Archive here.
- Please use gist or anything else to add other files and add links here

* ...

### PR DESCRIPTION
**Pull request by 95CivicSi**
PluginManager: Fix patch level check. to allow loading the plugin when the server's minor level is higher than the plugins declared minor level with matching majors.

## Introduction

The problem relevant to this PR is fully described in #2335 
### Relevant issues

* Fixes #2335 

### API changes

None
### Behavioural changes

Allows plugins to load predictably when the server major level is the same and minor level is higher than the plugins declared requirements.
## Tests

This was tested by changing the plugin API version of a test plugin on a 3.1.2dev instance.  
Results:
Plugin API 3.0.0 loaded
Plugin API 3.0.1 Loaded
Plugin API 3.0.2 Loaded
Plugin API 3.0.3 Loaded
Plugin API 3.0.14 Loaded
Plugin API 3.1.0 Loaded
Plugin API 3.1.1 Loaded
Plugin API 3.1.2 Loaded
Plugin API 3.1.3 Did not Load
Plugin API 3.1.14 Did not load.
Plugin API 3.2.0 Did not Load
Plugin API 3.2.1 Did not Load
Plugin API 3.2.2 Did not Load
Plugin API 3.2.3 Did not Load
Plugin API 3.2.14 Did not Load

### CODE REVIEW

File: src/pocketmine/plugin/PluginManager.php [hunk 1]
```diff
@@ -413,7 +413,7 @@ public function isCompatibleApi(string ...$versions) : bool{
 					continue;
 				}
 
-				if($pluginNumbers[2] > $serverNumbers[2]){ //If the plugin requires bug fixes in patches, being backwards compatible
+				if($pluginNumbers[2] > $serverNumbers[2] and $pluginNumbers[1] >= $serverNumbers[1]){ //If the plugin requires bug fixes in patches, being backwards compatible
```
Comment (SOF3) [inline src/pocketmine/plugin/PluginManager.php]: Code is confusing. `>=` could be simplified into `===`.

File: src/pocketmine/plugin/PluginManager.php [hunk 2]
```diff
@@ -413,7 +413,7 @@ public function isCompatibleApi(string ...$versions) : bool{
 					continue;
 				}
 
-				if($pluginNumbers[2] > $serverNumbers[2]){ //If the plugin requires bug fixes in patches, being backwards compatible
+				if($pluginNumbers[2] > $serverNumbers[2] and $pluginNumbers[1] === $serverNumbers[1]){ //If the plugin requires bug fixes in patches, being backwards compatible
```
Comment (dktapps) [inline src/pocketmine/plugin/PluginManager.php]: For clarity would probably be better to reverse the order of these checks.

Comment (95CivicSi) [reply src/pocketmine/plugin/PluginManager.php]: Ok.  This has been done.

**Comment by dktapps** [comment]
Merged to 3.0, 3.1, 3.2 and master in 4a1ed21e52b63b98235baec6c2d13ec04fab03d5.

**Pull request closed without merging**